code-analysis/crates/codegen/parser/src/chumsky/boilerplate.rs

```diff
@@ -17,41 +17,37 @@ pub fn lex_head() -> TokenStream {
 
         #[derive(Clone, Debug, PartialEq, Eq, Serialize)]
         pub enum Node {
-            None,
             Chars(Range<usize>),
             Choice(usize, Rc<Node>),
             Sequence(Vec<Rc<Node>>),
             Named(kinds::Token, Rc<Node>),
         }
 
         impl Node {
-            pub fn none() -> Rc<Self> {
-                Rc::new(Self::None)
+            pub fn chars(range: Range<usize>) -> Option<Rc<Self>> {
```

Comment:
Thinking out loud, is there going to be a semantic distinction for the `Node::chars()` API, or is it just a shorthand/utility? How is that different from `fn chars() -> RC<>` and `Some(chars(..))`?